### PR TITLE
Add jersey2 server support

### LINKED ISSUE #832: update jaxrs generator to jersey2
@olensmar any reason not to update the jaxrs to jersery2?

### PR DESCRIPTION
**Pull request by fehguy**
Adds templates for Jersey 2, updates formatting for Jersey 1.18 and makes generation timestamps optional.

Fixes #832, #1554, #2263

### CODE REVIEW

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractJavaJAXRSServerCodegen.java [hunk 1]
```diff
@@ -95,6 +96,25 @@ public void preprocessSwagger(Swagger swagger) {
             @SuppressWarnings("unchecked")
             List<CodegenOperation> ops = (List<CodegenOperation>) operations.get("operation");
             for ( CodegenOperation operation : ops ) {
+                boolean isMultipartPost = false;
```
Comment (fehguy) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractJavaJAXRSServerCodegen.java]: This addresses an issue where `@FormData` was being used for `multipart/form-data`, causing runtime errors when posting.

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaJerseyServerCodegen.java [hunk 2]
```diff
@@ -6,11 +6,10 @@
 import java.io.File;
 import java.util.*;
 
-public class JavaJerseyServerCodegen extends AbstractJavaJAXRSServerCodegen
-{
+public class JavaJerseyServerCodegen extends AbstractJavaJAXRSServerCodegen {
+    boolean showGenerationTimestamp = false;
```
Comment (fehguy) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaJerseyServerCodegen.java]: Made generation timestamps optional in server generation

File: bin/jersey2-petstore-server.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+executable="./modules/swagger-codegen-cli/target/swagger-codegen-cli.jar"
+
+if [ ! -f "$executable" ]
+then
+  mvn clean package
+fi
+
+# if you've executed sbt assembly previously it will use that instead.
+export JAVA_OPTS="${JAVA_OPTS} -XX:MaxPermSize=256M -Xmx1024M -DloggerPath=conf/log4j.properties"
+ags="$@ generate -t modules/swagger-codegen/src/main/resources/JavaJaxRS -i modules/swagger-codegen/src/test/resources/2_0/petstore.yaml -l jaxrs -o samples/server/petstore/jersey2 --library=jersey2"
```
Comment (fehguy) [inline bin/jersey2-petstore-server.sh]: changed to use `petstore.yaml` as it doesn't have any of the test constructs in it, and re-adds file upload support

**Comment by wing328** [comment]
@fehguy I ran some tests and didn't find any issues. PR merged.

**Pull request merged**